R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
@felixcheung sorry, I don't get what you mean.  L357 will override whatever `master` is, that's why I make the change after that, so that we can pass the correct master to sparkConfigMap